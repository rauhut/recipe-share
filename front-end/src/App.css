.App {
  text-align: center;
}

button {
  padding: 5px 10px;
  font-weight: bolder;
  border-radius: 0.3rem !important;
}

button:hover {
  cursor: pointer;
}

.button-secondary {
  background-color: #cc1414;
  border: none;
  color: white;
  margin: 5px;
  text-transform: uppercase;
}

.button-primary {
  background-color: #ebebeb;
  border: none;
  color: #cc1414;
  margin: 5px;
  transition-duration: 0.4s;
  text-transform: uppercase;
}
