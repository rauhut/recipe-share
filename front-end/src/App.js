import React, { Component } from "react";
import MainPage from "./components/MainPage/MainPage";
import SignIn from "./components/SignIn/SignIn";
import Register from "./components/Register/Register";
import NavBar from "./components/NavBar/NavBar";
import RecipePage from "./components/RecipePage/RecipePage";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import "./App.css";

const initialState = {
  isSignedIn: false,
  user: {
    id: "",
    name: "",
    email: "",
    recipes: "",
  },
};

class App extends Component {
  constructor() {
    super();
    this.state = initialState;
  }

  componentDidMount() {
    const token = window.sessionStorage.getItem("token");
    if (token) {
      fetch("http://localhost:3000/signin", {
        method: "post",
        headers: {
          "Content-Type": "application/json",
          Authorization: token,
        },
      })
        .then((response) => response.json())
        .then((data) => {
          if (data.user) {
            fetch("http://localhost:3000/profile", {
              method: "get",
              headers: {
                "Content-Type": "application/json",
                Authorization: token,
              },
            })
              .then((response) => response.json())
              .then((user) => {
                if (user && user.email) {
                  this.loadUser(user);
                  this.onRouteChange("mainPage");
                  this.setState({ isSignedIn: true });
                }
              });
          }
        })
        .catch(console.log);
    }
  }

  loadUser = (data) => {
    this.setState({
      user: {
        id: data._id,
        name: data.name,
        email: data.email,
        recipes: data.recipes,
      },
    });
  };

  onRouteChange = (route) => {
    if (route === "signout") {
      this.setState({ isSignedIn: false });
      this.setState({ initialState });
    } else if (route === "signingIn") {
      this.setState({ isSignedIn: true });
    }
  };

  toggleModal = () => {
    this.setState((prevState) => ({
      ...prevState,
      isProfileOpen: !prevState.isProfileOpen,
    }));
  };

  render() {
    const { isSignedIn, user } = this.state;
    return (
      <Router>
        <div className="App">
          <NavBar
            isSignedIn={isSignedIn}
            onRouteChange={this.onRouteChange}
            toggleModal={this.toggleModal}
            user={user}
          />
          <Switch>
            <Route
              path="/"
              exact
              render={(props) => (
                <MainPage {...props} isSignedIn={isSignedIn} user={user} />
              )}
            />
            <Route
              path="/signin"
              render={(props) => (
                <SignIn
                  {...props}
                  onRouteChange={this.onRouteChange}
                  loadUser={this.loadUser}
                />
              )}
            />
            <Route
              path="/register"
              render={(props) => (
                <Register
                  {...props}
                  onRouteChange={this.onRouteChange}
                  loadUser={this.loadUser}
                />
              )}
            />
            <Route path="/recipe/:id" component={RecipePage} />
          </Switch>
        </div>
      </Router>
    );
  }
}

export default App;
