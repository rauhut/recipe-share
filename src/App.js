import React, { Component } from "react";
import MainPage from "./components/MainPage/MainPage";
import SignIn from "./components/SignIn/SignIn";
import Register from "./components/Register/Register";
import NavBar from "./components/NavBar/NavBar";
import RecipePage from "./components/RecipePage/RecipePage";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import "./App.css";
import ResultsPage from "./components/ResultsPage/ResultsPage";

const initialState = {
  isSignedIn: false,
  user: {
    id: "",
    name: "",
    email: "",
    recipes: "",
  },
  searchEntry: "",
  filteredRecipes: [],
};

class App extends Component {
  constructor() {
    super();
    this.state = initialState;
  }

  componentDidMount() {
    const token = window.sessionStorage.getItem("token");
    if (token) {
      fetch("http://localhost:3000/signin", {
        method: "post",
        headers: {
          "Content-Type": "application/json",
          Authorization: token,
        },
      })
        .then((response) => response.json())
        .then((data) => {
          if (data.user) {
            fetch("http://localhost:3000/profile", {
              method: "get",
              headers: {
                "Content-Type": "application/json",
                Authorization: token,
              },
            })
              .then((response) => response.json())
              .then((user) => {
                if (user && user.email) {
                  this.loadUser(user);
                  this.onRouteChange("mainPage");
                  this.setState({ isSignedIn: true });
                }
              });
          }
        })
        .catch(console.log);
    }
  }

  loadUser = (data) => {
    this.setState({
      user: {
        id: data._id,
        name: data.name,
        email: data.email,
        recipes: data.recipes,
      },
    });
  };

  onRecipeSearch = (search) => {
    this.setState({ searchEntry: search });
  };

  onRouteChange = (route) => {
    if (route === "signout") {
      this.setState({ isSignedIn: false });
      this.setState({ initialState });
    } else if (route === "signingIn") {
      this.setState({ isSignedIn: true });
    }
  };

  toggleModal = () => {
    this.setState((prevState) => ({
      ...prevState,
      isProfileOpen: !prevState.isProfileOpen,
    }));
  };

  render() {
    const { isSignedIn, user, searchEntry } = this.state;
    return (
      <Router>
        <div className="App">
          <NavBar
            isSignedIn={isSignedIn}
            onRouteChange={this.onRouteChange}
            toggleModal={this.toggleModal}
            user={user}
            onRecipeSearch={this.onRecipeSearch}
          />
          <Switch>
            <Route
              path="/"
              exact
              render={(props) => (
                <MainPage {...props} isSignedIn={isSignedIn} user={user} />
              )}
            />
            <Route
              path="/signin"
              render={(props) => (
                <SignIn
                  {...props}
                  onRouteChange={this.onRouteChange}
                  loadUser={this.loadUser}
                />
              )}
            />
            <Route
              path="/register"
              render={(props) => (
                <Register
                  {...props}
                  onRouteChange={this.onRouteChange}
                  loadUser={this.loadUser}
                />
              )}
            />
            <Route path="/recipe/:id" component={RecipePage} />
            <Route
              path="/search"
              render={(props) => (
                <ResultsPage {...props} searchEntry={searchEntry} />
              )}
            />
          </Switch>
        </div>
      </Router>
    );
  }
}

export default App;
