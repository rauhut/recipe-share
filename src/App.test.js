import React from "react";
import { shallow } from "enzyme";
import App from "./App";

describe("App", () => {
  let wrapper;

  const initialState = {
    isSignedIn: false,
    user: {
      id: "",
      username: "",
      recipes: "",
    },
    searchEntry: "",
    filteredRecipes: [],
  };

  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it("renders page without crashing", () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("updates state on onRouteChange", () => {
    wrapper.instance().onRouteChange("signout");
    expect(wrapper.state("isSignedIn")).toEqual(false);
    expect(wrapper.state("initialState")).toEqual(initialState);
    wrapper.instance().onRouteChange("signingIn");
    expect(wrapper.state("isSignedIn")).toEqual(true);
  });

  it("updates search field on text input", () => {
    wrapper.instance().onRecipeSearch("test search");
    expect(wrapper.state("searchEntry")).toEqual("test search");
  });

  it("updates user on loadUser call", () => {
    const mockUserData = {
      _id: "123456789",
      username: "testuser",
      recipes: "",
    };
    wrapper.instance().loadUser(mockUserData);
    expect(wrapper.state("user")).toEqual({
      id: "123456789",
      username: "testuser",
      recipes: "",
    });
  });
});
