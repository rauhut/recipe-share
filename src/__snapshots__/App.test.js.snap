// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders page without crashing 1`] = `
"<BrowserRouter>
  <div className=\\"App\\">
    <Route path=\\"/\\" render={[Function: render]} />
    <Switch>
      <Route path=\\"/\\" exact={true} render={[Function: render]} />
      <Route path=\\"/signin\\" render={[Function: render]} />
      <Route path=\\"/register\\" render={[Function: render]} />
      <Route path=\\"/recipe/:id\\" render={[Function: render]} />
      <Route path=\\"/search\\" render={[Function: render]} />
    </Switch>
  </div>
</BrowserRouter>"
`;
